src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -1109,23 +1110,39 @@ private static int LocateLastFoundByte(Vector<byte> match)
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int LocateFirstFoundByte(ulong match)
         {
-            // Flag least significant power of two bit
-            var powerOfTwoFlag = match ^ (match - 1);
-            // Shift all powers of two into the high byte and extract
-            return (int)((powerOfTwoFlag * XorPowerOfTwoToHighByte) >> 57);
+            // TODO: Arm variants
+            if (Bmi1.X64.IsSupported)
+            {
+                return (int)(Bmi1.X64.TrailingZeroCount(match) >> 3);
+            }
+            else
+            {
+                // Flag least significant power of two bit
+                var powerOfTwoFlag = match ^ (match - 1);
+                // Shift all powers of two into the high byte and extract
+                return (int)((powerOfTwoFlag * XorPowerOfTwoToHighByte) >> 57);
+            }
         }
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         private static int LocateLastFoundByte(ulong match)
         {
-            // Find the most significant byte that has its highest bit set
-            int index = 7;
-            while ((long)match > 0)
+            // TODO: Arm variants
+            if (Lzcnt.X64.IsSupported)
             {
-                match = match << 8;
-                index--;
+                return 7 - (int)(Lzcnt.X64.LeadingZeroCount(match) >> 3);
```

Comment:
This yields a different result than the `else` case, e.g. `LocateLastFoundByte((ulong)10) // when Lzcnt.X64.IsSupported` is `0` and `!Lzcnt.X64.IsSupported` (`else` case) returns `-1`. Is it known/intentional?
Aside from this, the result of `Lzcnt.X64.LeadingZeroCount((ulong)10)` is 60<sub>10</sub>, shouldn't it be 56<sub>10</sub>?

Reply:
10 isn't a value that can be passed to the method. The bytes in `match` can only be 0x00 or 0xff